config/configgrpc/configgrpc.go

```diff
@@ -400,7 +400,11 @@ func contextWithClient(ctx context.Context, includeMetadata bool) context.Contex
 	}
 	if includeMetadata {
 		if md, ok := metadata.FromIncomingContext(ctx); ok {
-			cl.Metadata = client.NewMetadata(md.Copy())
+			copiedMD := md.Copy()
+			if len(md[client.MetadataHostName]) == 0 && len(md[":authority"]) > 0 {
```

Comment:
I see now what @bogdandrutu meant earlier. I agree with the approach taken here, even though we end up with the host information duplicated in the client.Info. I don't think this is a problem.